### PR TITLE
Sage 10 docs

### PR DESCRIPTION
**Pull request by alwaysblank**
This PR will add an initial set of docs for Sage 10.

This is largely just mirroring the Sage 9 structure and content with things changed and added as necessary. There are parts of Sage 10 I've used a lot more than others, so it's possible I've missed something significant. Comments/request/criticisms are, as aways, encouraged.

### CODE REVIEW

**Review by Log1x — APPROVED** [review]

**Comment by jokanane** [comment]
Thanks @alwaysblank, happy to see Sage 10 docs coming together!

While merging a theme from Sage 9 to 10, I read the diffs (for the most part), here a couple of comments:

1) Localisation:

There's a section with `yarn pot` that looks to be from Sage 9; this script is no longer needed, as Sage 10 has now:

```json
"translate:pot": "wp i18n make-pot . ./resources/languages/sage.pot --ignore-domain --include=\"app,resources/assets,resources/views\"",
```

Also:
```markdown
add_action('after_setup_theme', function () {
    load_theme_textdomain('sage', get_theme_file_path('/resources/lang'));
});
```

Please note that the directory name is now `/resources/languages`, which needs to be updated in the `load_theme_textdomain()` part to reflect the aforementioned script change.

2) So there's no concept of "routes" in Sage 10 for JavaScript? Would it make sense to add a sentence / paragraph to clarify this as many who have worked with Sage 9 may expect still seeing this? Maybe add a note about using a `if(is_page('slug'))` conditional with `wp_enqueue_script()`?

3) I'm also a little puzzled with not having a port setting for local dev, or have I completely missed it? I've only got a good months' worth of Trellis experience now, but I think Trellis will keep rolling ports if 3000, 3001, ... are already taken; however earlier with MAMP I really had to explicitly set the port for each site (in theme's config.json) to avoid the notorious browsersync reload loop. Also, I'm thinking if the port gets handed "dynamically" per what's taken and what's available, isn't there a chance that e.g. cookies written for http://localhost:3000 are applied to another site that happens to be run in that port on another occasion?

**Comment by Log1x** [comment]
>So there's no concept of "routes" in Sage 10 for JavaScript? Would it make sense to add a sentence / paragraph to clarify this as many who have worked with Sage 9 may expect still seeing this? Maybe add a note about using a if(is_page('slug')) conditional with wp_enqueue_script()

The routing in Sage 9 was effectively pointless outside of providing an opinionated project structure. 

Simply check for the existence of the selector, etc. that you are working with and you have effectively the same thing.

>I'm also a little puzzled with not having a port setting for local dev, or have I completely missed it?

You can find the Laravel Mix documentation on BrowserSync [here](https://laravel-mix.com/docs/5.0/browsersync) which shows an example of passing an option to `mix.browserSync() and links to the official BrowserSync [documentation](https://browsersync.io/docs/options/#option-port)

In this case, you'd be looking for:

```js
mix.browserSync({
  proxy: 'sage.test',
  port: 3001,
});
```

File: docs/sage/10.x/compatibility.md [hunk 1]
```diff
@@ -7,8 +7,8 @@ description: A list of currently known compatibility issues with any WordPress p
 ## Adding support for plugins
 
 - No current official support for WooCommerce, but there are a number of options from the community that may be helpful:
-    - https://github.com/ptrckvzn/sage-woocommerce
-    - https://github.com/generoi/sage-woocommerce
+    - [https://github.com/ptrckvzn/sage-woocommerce](https://github.com/ptrckvzn/sage-woocommerce)
+    - [https://github.com/generoi/sage-woocommerce](https://github.com/ptrckvzn/sage-woocommerce)
```
Comment (joshuafredrickson) [inline docs/sage/10.x/compatibility.md]: Quick fix. Link should be:

`[https://github.com/generoi/sage-woocommerce](https://github.com/generoi/sage-woocommerce)`

File: docs/sage/10.x/structure.md [hunk 2]
```diff
@@ -60,18 +60,18 @@ The `app` directory contains the core code and functionality of your theme. Almo
 
 ### The Config Directory
 
-The `config` directory contains configuration pertaining directly to Sage and it's functionality with Acorn. Here you will not find much confugration related to WordPress but instead configuration for things such as modifying Blade view paths, registering directives, changing the asset path, and configuration for packages built for Sage.
+The `config` directory contains configuration pertaining directly to Sage and it's functionality with Acorn. Here you will not find much configration related to WordPress but instead configuration for things such as modifying Blade view paths, registering directives, changing the asset path, and configuration for packages built for Sage.
```
Comment (joshuafredrickson) [inline docs/sage/10.x/structure.md:63]: `configuration`

**Review by retlehs — APPROVED** [review]

**Pull request merged**